vms/platformvm/txs/fee/complexity_test.go

```diff
@@ -0,0 +1,543 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fee
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/components/fee"
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/platformvm/fx"
+	"github.com/ava-labs/avalanchego/vms/platformvm/signer"
+	"github.com/ava-labs/avalanchego/vms/platformvm/stakeable"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+)
+
+func TestTxComplexity(t *testing.T) {
+	for _, test := range txTests {
+		t.Run(test.name, func(t *testing.T) {
+			require := require.New(t)
+
+			txBytes, err := hex.DecodeString(test.tx)
+			require.NoError(err)
+
+			tx, err := txs.Parse(txs.Codec, txBytes)
+			require.NoError(err)
+
+			// If the test fails, logging the transaction can be helpful for
+			// debugging.
+			txJSON, err := json.MarshalIndent(tx, "", "\t")
+			require.NoError(err)
+			t.Log(string(txJSON))
+
+			actual, err := TxComplexity(tx.Unsigned)
+			require.Equal(test.expectedComplexity, actual)
+			require.ErrorIs(err, test.expectedComplexityErr)
+			if err != nil {
+				return
+			}
+
+			require.Len(txBytes, int(actual[fee.Bandwidth]))
+		})
+	}
+}
+
+func BenchmarkTxComplexity(b *testing.B) {
+	for _, test := range txTests {
+		b.Run(test.name, func(b *testing.B) {
+			require := require.New(b)
+
+			txBytes, err := hex.DecodeString(test.tx)
+			require.NoError(err)
+
+			tx, err := txs.Parse(txs.Codec, txBytes)
+			require.NoError(err)
+
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				_, _ = TxComplexity(tx.Unsigned)
+			}
+		})
+	}
+}
+
+func TestOutputComplexity(t *testing.T) {
+	tests := []struct {
+		name        string
+		out         *avax.TransferableOutput
+		expected    fee.Dimensions
+		expectedErr error
+	}{
+		{
+			name: "any can spend",
+			out: &avax.TransferableOutput{
+				Out: &secp256k1fx.TransferOutput{
+					OutputOwners: secp256k1fx.OutputOwners{
+						Addrs: make([]ids.ShortID, 0),
+					},
+				},
+			},
+			expected: fee.Dimensions{
+				fee.Bandwidth: 60,
```

Comment:
The expected bandwidth fee is actually explicitly checked by the codec serialization. We could populate this dynamically in the test... But I felt like it made more sense to have the value specified and checked in the test... Do you think we should just remove the specification of the value in the vector?

Reply:
I added a comment to clarify the definition. Because this field is just the length of the tx in bytes - I'm not sure it needs too much additional context